SFML/main.cpp

```diff
@@ -0,0 +1,243 @@
+
+#include <SFML/Graphics.hpp>
+#include <SFML/Window.hpp>
+#include <SFML/System.hpp>
+#include <filesystem>
+#include <iostream>
+#include "Util.h"
+
+static sf::RenderWindow window;
+
+class Boundaries
+{
+public:
+	float leftBoundary;
+	float rightBoundary;
+	float topBoundary;
+	float bottomBoundary;
+	float thickness;
+
+	sf::RectangleShape topRect;
+	sf::RectangleShape bottomRect;
+	sf::RectangleShape leftRect;
+	sf::RectangleShape rightRect;
+
+	bool isBoundaryCollisions = false;
+
+	Boundaries(sf::RenderWindow& window, float left, float right, float top, float bottom)
+	{
+		this->leftBoundary = left;
+		this->rightBoundary = window.getSize().x - right;
+		this->topBoundary = top;
+		this->bottomBoundary = window.getSize().y - bottom;
+		this->thickness = 10.0f;
+	}
+
+	void draw(sf::RenderWindow& window)
+	{
+		topRect.setSize(sf::Vector2f(rightBoundary - leftBoundary, thickness));
+		topRect.setPosition(leftBoundary, topBoundary);
+		topRect.setFillColor(sf::Color::Red);
+
+		bottomRect.setSize(sf::Vector2f(rightBoundary - leftBoundary, thickness));
+		bottomRect.setPosition(leftBoundary, bottomBoundary - thickness);
+		bottomRect.setFillColor(sf::Color::Red);
+
+		leftRect.setSize(sf::Vector2f(thickness, bottomBoundary - topBoundary));
+		leftRect.setPosition(leftBoundary, topBoundary);
+		leftRect.setFillColor(sf::Color::Red);
+
+		rightRect.setSize(sf::Vector2f(thickness, bottomBoundary - topBoundary));
+		rightRect.setPosition(rightBoundary - thickness, topBoundary);
+		rightRect.setFillColor(sf::Color::Red);
+
+		window.draw(topRect);
+		window.draw(bottomRect);
+		window.draw(leftRect);
+		window.draw(rightRect);
+
+	}
+
+	bool checkBoundariesCollision(sf::Sprite& object)
+	{
+		isBoundaryCollisions = false;
+
+		sf::FloatRect globalBounds = object.getGlobalBounds();
+		sf::FloatRect topRectBounds = topRect.getGlobalBounds();
+
+		if (Util::checkCollision(globalBounds, topRect.getGlobalBounds())
+			|| Util::checkCollision(globalBounds, bottomRect.getGlobalBounds())
+			|| Util::checkCollision(globalBounds, leftRect.getGlobalBounds())
+			|| Util::checkCollision(globalBounds, rightRect.getGlobalBounds()))
+		{
+			isBoundaryCollisions = true;
+		}
+
+		return isBoundaryCollisions;
+	}
+
+	void handleMovement(sf::Sprite& gameObject)
+	{
+		float DeltaX = 0.0f;
+		float DeltaY = 0.0f;
+		if (sf::Keyboard::isKeyPressed(sf::Keyboard::Key::W))
+		{
+			DeltaY = -10.0f;
+		}
+		else if (sf::Keyboard::isKeyPressed(sf::Keyboard::Key::S))
+		{
+			DeltaY = 10.0f;
+		}
+		else if (sf::Keyboard::isKeyPressed(sf::Keyboard::Key::D))
+		{
+			DeltaX = 10.0f;
+		}
+		else if (sf::Keyboard::isKeyPressed(sf::Keyboard::Key::A))
+		{
+			DeltaX = -10.0f;
+		}
+		else
+		{
+			return;
+		}
+
+		const sf::Vector2f oldPosition = gameObject.getPosition();
+		gameObject.setPosition(oldPosition.x + DeltaX, oldPosition.y + DeltaY);
+
+		if (checkBoundariesCollision(gameObject))
+		{
+			gameObject.setPosition(oldPosition);
+		}
```

Comment:
may be the case when you can move by one axis but not other. E.g. ok to move by X and not by Y but you restrict both